### ISSUE TITLE
fetch all record from dynamoDB using graphql

### ISSUE THREAD

**Issue by prabha-912** [aws-amplify/amplify-js#7876]
**Describe the bug**
I'm having 150+ records in table. While I'm trying to fetch the records based on the user, some records are missed. 

In the 150 records, X user having 10 records in first 100 (page 1), 5 records in second 50 (page 2). When I try to get X's 15 record, only 10 records are returned. we missed 5 records.

**To Reproduce**

1. Have paginated records in dynamoDB. 
2. list the table data using AppSync based on the user filter.
3. Count the items. It will return the first page data. Not from second page.

**Expected behavior**
We have to get all the records for the user X. 

We used nextToken method. But the flow of the next Token is, 

If my record totally saved on second page, then it returns empty set. Its not a problem for  100, 200. If the table data goes high, multiple loops needs to be run. 

Without next Token concept or looping concept, can we achieve this data fetching? We need a best practice to use.

**Code Snippet**
`query MyQuery {
  listCustomData {
    items {
      title
    }
  }
}`

Sample Output:
It returns, 10 data, but totally I'm having 15 data in my name.

> {
  "data": {
    "listCustomData": {
      "items": [
        {
          "title": "title-1"
        },
        {
          "title": "title-2"
        },
        ...

**Comment by nubpro** [comment]
The problem with `listQuery`:
It performs a scan on the entire DB (which means the entire 150 records).
BUT, there's a default limit. Lets say it's 10 (idk what the real number).
So instead of paging through the entire 150 records, it is capped at 10. Which means, if all X user's records lives beyond the 10 records, nothing will be returned.
This is because your filter expression come after the limiting procedure.

In short:
Scan -> Limit -> Filter

NOT:
Scan -> Filter -> Limit

Here's what I'm proposing:
1. Dont use the default `listQuery` generated.
2. Setup a GSI. The index would be `userId`.
```
type Data @model
@key(name: "byUser", fields: ["userId"], queryField: "dataByUser")
```
3. With that, you can now query using `dataByUser`, you need to include the user's id as the request input.

**Comment by prabha-912** [comment]
Hi @nubpro, Thanks for your quick response, I'll try with this suggested solution and will post the update here. Thanks much again.

**Comment by prabha-912** [comment]
Yes, we can achieve this by the recommended solution. Thanks much @nubpro. It's working fine as expected.
